format/FlightSql.proto

```diff
@@ -867,6 +867,151 @@ enum SqlSupportsConvert {
   SQL_CONVERT_VARCHAR = 19;
 }
 
+enum DATE_SUBCODE {
+  UNKNOWN_SUBCODE = 0;
```

Comment:
Thank you for adding an UNKNOWN case - that said, can we keep it (and the other values) named consistently? This should be SQL_DATE_SUBCODE_UNKNOWN (and the others should be SQL_DATE_CODE_DATE, etc.)

Reply:
Or I guess `SQL_DATETIME_SUBCODE_UNKNOWN`  since the field is `datetime_subcode` below.